### PR TITLE
AUTH FOR HOMEPAGE WOOH.

### PR DESCRIPTION
**Pull request by janssejl**

### CODE REVIEW

File: _assets/javascripts/lib/authentication.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+let oktaConfig = {
+  clientId: CRDS.env.okta_client_id,
```
Comment (j-brammer) [inline _assets/javascripts/lib/authentication.js]: Is there a reason we are declaring clientId and issuer using {site.XXX} on auth-required and using CRDS.env here?

File: _assets/javascripts/lib/authentication.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    crdsAuth.CrdsAuthenticationProviders.Okta,
+    crdsAuth.CrdsAuthenticationProviders.Mp
+  ]
+}
+
+let authService = new crdsAuth.CrdsAuthenticationService(authConfig);
+
+authService.authenticated().subscribe(tokens => {
+  if (tokens)
+    document.getElementsByTagName("crds-site-happenings")[0].setAttribute("auth-token", tokens.access_token.access_token);
```
Comment (lukemoderwell) [inline _assets/javascripts/lib/authentication.js]: A consideration here: Maybe look for that prop in general and then update all components using this concept.

```
let authEls = document.querySelectorAll('[auth-token]')
authEls.forEach(...set this attr)
```

File: _assets/javascripts/lib/authentication.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  providerPreference: [
+    // Put these in the order you want the library to check for auth status
+    crdsAuth.CrdsAuthenticationProviders.Okta,
+    crdsAuth.CrdsAuthenticationProviders.Mp
+  ]
+}
+
+let authService = new crdsAuth.CrdsAuthenticationService(authConfig);
+
+authService.authenticated().subscribe(tokens => {
```
Comment (j-brammer) [inline _assets/javascripts/lib/authentication.js]: Seems like the only difference in authentication.js and auth-required.js is this little block of code here. 

Could we maybe key off the "auth required" contentful page attribute to decide which type of auth (full page required, or just component required) to use?

File: _assets/javascripts/lib/authentication.js (hunk 2 above)
Comment (janssejl) [reply _assets/javascripts/lib/authentication.js]: done

File: _assets/javascripts/lib/authentication.js (hunk 3 above)
Comment (janssejl) [reply _assets/javascripts/lib/authentication.js]: refactored.

File: _assets/javascripts/lib/authentication.js (hunk 1 above)
Comment (janssejl) [reply _assets/javascripts/lib/authentication.js]: just goofed on not changing that file from {site.....} when I realized that didnt work. Refactored anyway.

**Pull request merged**